### PR TITLE
Change "stack" terminology to "queue"

### PR DESCRIPTION
**Pull request by pittst3r**
## Purpose

Addresses #9.

The term "queue" more accurately describes the usage of the array in the `Convergence` class used to store callbacks and assertions. We do two things with it: append items to the end and iterate; making it first-in-first-out.

## Approach

Wish I could say that I used `find` and `sed` and so forth but I am not a h4x0r and used VSCode find and replace instead.

I also added a test expressing this aspect of the API.

### CODE REVIEW

File: tests/convergence-test.js
Comment (wwilsman) [inline tests/convergence-test.js]: The other tests here are written in a specific way, so we should try to be consistent when adding new ones. The main thing is: we don't want to put the real assertions in the convergence to ensure that we don't get a false positive in the off chance that assertion doesn't get called.

We also don't need `.when(() => true)` here, and can use `.to.deep.equal` instead of joining the array.

```js
it('stores and calls the callbacks first-in-first-out', async () => {
  let arr = [];
  
  let assertion = converge
    .do(() => arr.push('first'))
    .do(() => arr.push('second'));

  await expect(assertion.run()).to.be.fulfilled;
  expect(arr).to.deep.equal(['first', 'second']);
});
```

**Review by wwilsman — CHANGES_REQUESTED** [review]
One small nit about the test, but looks good!

**Review by Robdel12 — APPROVED** [review]
This looks good to me! Thanks for the PR 🎉 😎

File: tests/convergence-test.js
Comment (pittst3r) [reply tests/convergence-test.js]: A++++. So much better.

Comment (pittst3r) [reply tests/convergence-test.js]: K, updated. 😃

**Review by wwilsman — APPROVED** [review]

**Comment by Robdel12** [comment]
@robbiepitts we usually leave it to the dev to merge their own PRs when they're ready. So merge when you're ready 😀

**Comment by pittst3r** [comment]
Cool, good to know. Will do that, but don't have those permissions to atm.

**Comment by Robdel12** [comment]
Huh, I wonder if that's because this is a fork and not a branch on the repo? You're a member of the org

**Comment by pittst3r** [comment]
Well, there are other things I can't do like edit the repo summary line or add tags to issues. I'm seeing this, which makes me think maybe members don't get write access automatically or something like that:

screen shot 2018-05-03 at 10 10 26 am

**Pull request merged**
